nibabel/testing/__init__.py

```diff
@@ -56,22 +52,50 @@ def assert_allclose_safely(a, b, match_nans=True):
     assert_true(np.allclose(a, b))
 
 
-class suppress_warnings(catch_warnings):
-    """ Version of ``catch_warnings`` class that suppresses warnings
-    """
-    def __enter__(self):
-        res = super(suppress_warnings, self).__enter__()
-        simplefilter('ignore')
-        return res
+class clear_and_catch_warnings(warnings.catch_warnings):
```

Comment:
Note: I just changed the default to `record=True` for the nibabel version of this code. This is what was used in all of these helper classes in the `nibabel` codebase previously, and I think that makes a lot of sense.